### PR TITLE
Add environment variable to print driver info in VLTs

### PR DESCRIPTION
**Pull request by Tony-LunarG**
At the request of our internal CI team

### CODE REVIEW

**Comment by Tony-LunarG** [comment]
Everyone ok with this?

File: tests/vkrenderframework.cpp [hunk 1]
```diff
@@ -423,6 +423,7 @@ VkInstanceCreateInfo VkRenderFramework::GetInstanceCreateInfo() const {
     };
 }
 
+static bool driver_printed = false;
```
Comment (jeremyg-lunarg) [inline tests/vkrenderframework.cpp:426]: can this be moved inside the method?

**Review by jeremyg-lunarg — CHANGES_REQUESTED** [review]

File: tests/vkrenderframework.cpp [hunk 2]
```diff
@@ -470,6 +471,18 @@ void VkRenderFramework::InitFramework(void * /*unused compatibility parameter*/,
 
     vk::GetPhysicalDeviceProperties(gpu_, &physDevProps_);
     debug_reporter_.Create(instance_);
+
+    if (getenv("VK_LAYER_TESTS_PRINT_DRIVER") && !driver_printed) {
```
Comment (jeremyg-lunarg) [inline tests/vkrenderframework.cpp:475]: should this be GetEnvironment()? (vk_layer_config.h)

File: tests/vkrenderframework.cpp [hunk 3]
```diff
@@ -470,6 +471,18 @@ void VkRenderFramework::InitFramework(void * /*unused compatibility parameter*/,
...
+        VkPhysicalDeviceDriverProperties driver_properties = {};
+        VkPhysicalDeviceProperties2 physical_device_properties2 = {};
+        physical_device_properties2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2;
+        physical_device_properties2.pNext = &driver_properties;
+        driver_properties.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DRIVER_PROPERTIES;
+        vk::GetPhysicalDeviceProperties2(gpu_, &physical_device_properties2);
+        printf("Driver Name = %s\n", driver_properties.driverName);
```
Comment (jeremyg-lunarg) [inline tests/vkrenderframework.cpp:482]: Will this work correctly on android? LOG_CONSOLE() instead?

File: tests/vkrenderframework.cpp (hunk 2 above)
Comment (ncesario-lunarg) [inline tests/vkrenderframework.cpp:475]: Should this be `GetEnvironment` due to inconsistencies across platforms? That said, this probably won't be useful on Android, which I think is the only place this wouldn't work.

If it does need to be `GetEnvironment`, I think the tests executable would have to link against `VkLayer_utils`.

File: tests/vkrenderframework.cpp [hunk 4]
```diff
@@ -470,6 +471,18 @@ void VkRenderFramework::InitFramework(void * /*unused compatibility parameter*/,
...
     vk::GetPhysicalDeviceProperties(gpu_, &physDevProps_);
     debug_reporter_.Create(instance_);
+
+    if (getenv("VK_LAYER_TESTS_PRINT_DRIVER") && !driver_printed) {
+        VkPhysicalDeviceDriverProperties driver_properties = {};
+        VkPhysicalDeviceProperties2 physical_device_properties2 = {};
+        physical_device_properties2.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_PROPERTIES_2;
+        physical_device_properties2.pNext = &driver_properties;
+        driver_properties.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_DRIVER_PROPERTIES;
+        vk::GetPhysicalDeviceProperties2(gpu_, &physical_device_properties2);
```
Comment (ncesario-lunarg) [inline tests/vkrenderframework.cpp:481]: Will this be a problem for "pre 1.1" devices?

**Review by ncesario-lunarg — COMMENTED** [review]
Couple clarifying questions, but in general LGTM!

File: tests/vkrenderframework.cpp (hunk 3 above)
Comment (jeremyg-lunarg) [reply tests/vkrenderframework.cpp:482]: NVM on this one

File: tests/vkrenderframework.cpp (hunk 1 above)
Comment (Tony-LunarG) [reply tests/vkrenderframework.cpp]: Yes

File: tests/vkrenderframework.cpp (hunk 2 above)
Comment (Tony-LunarG) [reply tests/vkrenderframework.cpp]: Yes

Comment (Tony-LunarG) [reply tests/vkrenderframework.cpp]: Yes

File: tests/vkrenderframework.cpp (hunk 4 above)
Comment (Tony-LunarG) [reply tests/vkrenderframework.cpp]: Yes, but I believe I've fixed it now

File: tests/vkrenderframework.cpp [hunk 5]
```diff
@@ -470,6 +470,21 @@ void VkRenderFramework::InitFramework(void * /*unused compatibility parameter*/,
 
     vk::GetPhysicalDeviceProperties(gpu_, &physDevProps_);
     debug_reporter_.Create(instance_);
+
+    static bool driver_printed = false;
+    if (GetEnvironment("VK_LAYER_TESTS_PRINT_DRIVER") != "" && !driver_printed) {
+        if (InstanceExtensionSupported(VK_KHR_GET_PHYSICAL_DEVICE_PROPERTIES_2_EXTENSION_NAME)) {
+            VkPhysicalDeviceDriverProperties driver_properties = {};
+            VkPhysicalDeviceProperties2 physical_device_properties2 = {};
```
Comment (ncesario-lunarg) [inline tests/vkrenderframework.cpp:478]: Could collapse this and the next 2 lines into a single `LvlInitStruct` call if you want.

**Review by ncesario-lunarg — APPROVED** [review]
Couple minor suggestions, but LGTM!

**Pull request merged**